programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -293,16 +378,68 @@ static bool process_tunnel_commands(const tunnel_comand *tnl_cmd, command_cb cb,
             result.data = tunnel_status_to_json(status, MODEL_JSON_COMPACT, &json_len);
             break;
         }
+
+        case TunnelCommand_AddIdentity : {
+            cmd_accepted = true;
+            tunnel_add_identity tunnel_add_identity_cmd = {0};
+            if (tnl_cmd->data == NULL ||
+                parse_tunnel_add_identity(&tunnel_add_identity_cmd, tnl_cmd->data, strlen(tnl_cmd->data)) < 0) {
+                result.error = "invalid command";
+                result.success = false;
+                free_tunnel_add_identity(&tunnel_add_identity_cmd);
+                break;
+            }
+
+            char* extension = strstr(tunnel_add_identity_cmd.jwtFileName, ".jwt");
+            int length;
+            if (extension != NULL) {
+                length = (int) (extension - tunnel_add_identity_cmd.jwtFileName);
+            } else {
+                length = strlen(tunnel_add_identity_cmd.jwtFileName);
+            }
+            char new_identifier[FILENAME_MAX];
+            char new_identifier_name[FILENAME_MAX];
+            memcpy(&new_identifier_name, tunnel_add_identity_cmd.jwtFileName, length);
+            new_identifier_name[length] = '\0';
+            sprintf(new_identifier, "%s/%s.json", config_dir, new_identifier_name);
+            FILE *outfile;
+            if ((outfile = fopen(new_identifier, "wb")) == NULL) {
+                ZITI_LOG(ERROR, "failed to open file %s: %s(%d)", new_identifier, strerror(errno), errno);
+                result.error = "invalid file name";
+                result.success = false;
+                free_tunnel_add_identity(&tunnel_add_identity_cmd);
+                break;
+            }
+
+            struct add_identity_request_s *add_id_req = calloc(1, sizeof(struct add_identity_request_s));
```

Comment:
this is leaked

Reply:
add_id_req data is free'd in the tunnel_enroll_cb callback function.